### PR TITLE
soroban-rpc: Fix set_authorization_entries bug in preflight

### PR DESCRIPTION
**Pull request by tamirms**
### What

Configure the host source account before calling `host.set_authorization_entries()`.

### Why

`host.set_authorization_entries()` will fail if the source account is not set in the source. This is the specific error which will trigger:

https://github.com/stellar/rs-soroban-env/blob/main/soroban-env-host/src/auth.rs#L1366

### Known limitations

[N/A]

### CODE REVIEW

File: cmd/soroban-rpc/lib/preflight/src/preflight.rs [hunk 1]
```diff
@@ -63,13 +69,6 @@ pub(crate) fn preflight_invoke_hf_op(
             .context("cannot set authorization entries")?;
```
[the remark this answers is not in the document]
Comment (2opremio) [reply cmd/soroban-rpc/lib/preflight/src/preflight.rs:69]: Ahhh, this is also why we weren't getting debug info!

[the remark this answers is not in the document]
Comment (2opremio) [reply cmd/soroban-rpc/lib/preflight/src/preflight.rs:69]: Good catch!

**Review by 2opremio — APPROVED** [review]

File: cmd/soroban-rpc/lib/preflight/src/preflight.rs [hunk 2]
```diff
@@ -46,6 +46,12 @@ pub(crate) fn preflight_invoke_hf_op(
         .context("cannot create budget")?;
     let storage = Storage::with_recording_footprint(ledger_storage_rc.clone());
     let host = Host::with_storage_and_budget(storage, budget);
+    host.set_source_account(source_account.clone())
```
Comment (tsachiherman) [inline cmd/soroban-rpc/lib/preflight/src/preflight.rs:49]: The source account passed in here is either the operation source account or the transaction source account, right ? and that logic aligns with the core's implementation, right ? @sisuresh could you confirm that ?

**Review by tsachiherman — APPROVED** [review]

File: cmd/soroban-rpc/lib/preflight/src/preflight.rs (hunk 2 above)
Comment (sisuresh) [reply cmd/soroban-rpc/lib/preflight/src/preflight.rs:49]: Yeah that aligns with what core passes to the host - https://github.com/stellar/stellar-core/blob/23f3cfb821d79f3958b615e0404d6f6ff446cdaa/src/transactions/OperationFrame.cpp#L220.

File: cmd/soroban-rpc/internal/test/simulate_transaction_test.go [hunk 3]
```diff
@@ -306,6 +306,64 @@ func TestSimulateTransactionSucceeds(t *testing.T) {
...
+	assert.NoError(t, xdr.SafeUnmarshalBase64(response.Results[0].Auth[0], &auth))
+	require.Equal(t, auth.Credentials.Type, xdr.SorobanCredentialsTypeSorobanCredentialsSourceAccount)
+	deployContractOp.Auth = append(deployContractOp.Auth, auth)
+	deployContractParams.Operations = []txnbuild.Operation{deployContractOp}
+
+	// preflight deployContractOp with auth
+	deployContractParams = preflightTransactionParams(t, client, deployContractParams)
+	tx, err = txnbuild.NewTransaction(deployContractParams)
+	assert.NoError(t, err)
+	sendSuccessfulTransaction(t, client, sourceAccount, tx)
```
Comment (sreuland) [inline cmd/soroban-rpc/internal/test/simulate_transaction_test.go:364]: was kind of expecting this to re-run simulation with the tx now having the auth present?

```
response := simulateTransactionFromTxParams(t, client, deployContractParams)
```

Comment (tsachiherman) [reply cmd/soroban-rpc/internal/test/simulate_transaction_test.go:364]: why ? rerunning the simulateTransaction is only needed in the case of non-source account transaction ( i.e. where you need other address to sign it )

Comment (sreuland) [reply cmd/soroban-rpc/internal/test/simulate_transaction_test.go:364]: ah ok, cool, yeah, this test is essentially replicating the same rpc call flow that cli does during `contract deploy` for coverage.

**Pull request merged**